esrally/mechanic/supplier.py

```diff
@@ -243,7 +243,8 @@ def __init__(self, version, os_name=None, arch=None):
         if arch is not None:
             self.arch = arch
         else:
-            self.arch = sysstats.cpu_arch().lower()
+            derived_arch = sysstats.cpu_arch().lower()
+            self.arch = "aarch64" if derived_arch == "arm64" else derived_arch
```

Comment:
Maybe a small comment would explain this confusing line?

```suggestion
            derived_arch = sysstats.cpu_arch().lower()
            # Elasticsearch artifacts for Apple Silicon use "aarch64" as the CPU architecture
            self.arch = "aarch64" if derived_arch == "arm64" else derived_arch
```

Reply:
Isn't it the other way around, i.e. Linux reports `aarch64` but Apple `arm64`?

Reply:
Apple does report `arm64`, but the Apple Silicon URL is `https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-7.16.3-darwin-aarch64.tar.gz`, which is why Rally needs to use `aarch64` when the OS reports `arm64`

Reply:
Thanks for the suggestion. I've committed your comment as well. Once CI is green, I'll merge it.